doc/api/esm.md

```diff
@@ -1506,9 +1516,9 @@ _isImports_, _conditions_)
 > 7. If _pjson?.type_ exists and is _"module"_, then
 >    1. If _url_ ends in _".js"_, then
 >       1. Return _"module"_.
->    2. Throw an _Unsupported File Extension_ error.
+>    2. return **undefined**.
```

Comment:
When `ESM_FILE_FORMAT` is called we then set this value to the `format` variable.

It could be worth also changing the name here from `format` to `defaultFormat` so it is clearer that load phase overrides are permitted.

Reply:
Not sure what `format` you were referring to.

If it was to the "variable" in the resolution algorithm, then I think we should leave it `format` as the resolution algorithm is not really the whole resolution algorithm (which should detail how it handles loaders), but only the Node.js default resolved. In that case, making everything "default" is not needed. 

I _did_ change the titles of the main sections here from `Resolver...` to `Default resolver ...`

Reply:
The important point here is that the `format` at resolution time is a "hint", while the `format` at load time is authoritative. That is load-time can always override the resolve-time format, and this is how the loader pipeline now works (it wasn't originally). So noting this in the algorithm, since we are returning `format` as a being authoritative in the current resolution algorithm, that this PR is changing format to no longer being authoritative but only the suggested format which can then be further processed at load time.

Reply:
Got it! Will think of a way to incorporate this